host/xtest/regression_1000.c

```diff
@@ -2349,3 +2349,239 @@ static void xtest_tee_test_1032(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(regression, 1032, xtest_tee_test_1032,
 		"Register read-only shared memory");
+
+/*
+ * regression_1033
+ * Depends on kernel commit "tee: add support for application-based session login methods"
+ * Mailing list archives: https://lists.trustedfirmware.org/pipermail/op-tee/2020-October/000211.html
+ * Upstream: <put sha-1 here when known>
+ *
+ * xtest skips the test when not built with OpenSSL.
+ */
+static void xtest_tee_test_1033(ADBG_Case_t *c)
+{
+#ifdef OPENSSL_FOUND
+	TEEC_Result res = TEEC_ERROR_GENERIC;
+	uint32_t ret_orig = 0;
+	TEEC_Session session = { };
+	uint32_t login = UINT32_MAX;
+	TEEC_UUID client_uuid = { };
+	TEEC_UUID expected_client_uuid = { };
+	TEEC_UUID uuid_ns = { };
+	char uuid_name[TEE_UUID_NS_NAME_SIZE + 4] = { };
+	char app[PATH_MAX] = { };
+	int len = 0;
+
+	res = xtest_uuid_from_str(&uuid_ns, client_uuid_linux_ns);
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, res))
+		return;
+
+	if (readlink("/proc/self/exe", app, PATH_MAX) < 0) {
```

Comment:
In the kernel there are two methods for matching -- this requires that the kernel is configured for CONFIG_TEE_APPID_PATH.

If this is configured differently then the test will fail which is of course the then what we want.

In theory we could get it from /proc/config.gz IF that is available on running system -- but that might not be the case. 

Any ideas how we should handle this?

Reply:
We already had similar cases where optee components config/version could affect xtest result.
For those, we rather suggested in discussions (I don't remember which) to disable the test case (i.e. here `xtest -x 1033`) when running latest optee_test xtest with older optee components.